mne/decoding/base.py

```diff
@@ -746,3 +746,98 @@ def _make_scorer(scoring):
         return get_scorer(scoring)
     else:
         return make_scorer(scoring)
+
+
+def dummy_encoding(epochs, use_events=None):
+    """Create decoding targets based on epoch event codes.
+
+    Creates (0/1) integer values suitable for use as a target variable (y) for
+    decoding.
+
+    If two classes are present in the data and each epoch belongs exclusively
+    to either class, a list will be generated where 0 denotes that the epoch
+    belongs to the first class, and a 1 denotes that the epoch belongs to the
+    second class.
+
+    If more than two classes are present, or epochs are present that belong to
+    no class or to multiple classes, a matrix will be generated where each
+    column corresponds to a class and each row encodes which class(es) the
+    epoch belongs to. By default, the columns of the matrix will correspond to
+    the events in order of their respective event-ids (further ordered by their
+    event name if multiple classes share the same event-id). Use the
+    `use_events` parameter to specify a different order.
+
+    Parameters
+    ----------
+    epochs : instance of :class:`mne.Epochs`
+        The epochs to create the dummy encoding for.
+    use_events : list of str | None
```

Comment:
it's not the same because you lose the ordering if you do it, no? I am not sure ...